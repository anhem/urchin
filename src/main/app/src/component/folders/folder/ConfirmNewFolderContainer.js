import React, {Component} from 'react'
import {connect} from 'react-redux'
import ConfirmNewFolder from './ConfirmNewFolder';
import {confirmEncryptedFolder, setEncryptedFolderPassphrase} from '../../../action/folderAction';

export class ConfirmNewFolderContainer extends Component {

    render() {
        return (
            <ConfirmNewFolder confirmNewFolder={this.props.confirmNewFolder}
                              createdFolder={this.props.createdFolder}
                              fieldErrors={this.props.fieldErrors}
                              callbacks={{
                                  setEncryptedFolderPassphrase: this.props.setEncryptedFolderPassphrase,
                                  confirmEncryptedFolder: this.props.confirmEncryptedFolder
                              }}/>
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        setEncryptedFolderPassphrase: (passphrase) => {
            dispatch(setEncryptedFolderPassphrase(passphrase))
        },
        confirmEncryptedFolder: (folderId, passphrase) => {
            dispatch(confirmEncryptedFolder(folderId, passphrase))
        }
    }
};

const mapStateToProps = (state) => {
    return {
        confirmNewFolder: state.folderData.confirmNewFolder || {},
        createdFolder: state.folderData.createdFolder,
        fieldErrors: state.folderData.fieldErrors || {}
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(ConfirmNewFolderContainer)