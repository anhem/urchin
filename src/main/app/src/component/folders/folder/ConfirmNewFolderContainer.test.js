import React from 'react';
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json';
import {ConfirmNewFolderContainer} from "./ConfirmNewFolderContainer";

describe('ConfirmNewFolderContainer', () => {

    it('match snapshot', () => {
        expect(toJson(shallow(<ConfirmNewFolderContainer/>))).toMatchSnapshot();
    });
});
