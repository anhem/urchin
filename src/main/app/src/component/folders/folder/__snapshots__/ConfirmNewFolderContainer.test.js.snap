// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmNewFolderContainer match snapshot 1`] = `
<ConfirmNewFolder
  callbacks={
    Object {
      "confirmEncryptedFolder": undefined,
      "setEncryptedFolderPassphrase": undefined,
    }
  }
/>
`;
